app/repository/RepositoryImplementation.java

```diff
@@ -44,14 +44,13 @@ public void delete(T model) {
         JPA.em().flush();
     }
 
-    public Criteria getCriteria() {
+    public Criteria getCriteria(Class c) {
```

Comment:
Why is this changed? It should work like it was, `getCriteria` should use class resolved in `getEntityClass`.